test/fileutils.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/fileutils.hpp>
+#include <test.hpp>
+
+namespace fs = migraphx::fs;
+
+// NOLINTBEGIN
+// Explicit use of #define to concatenate strings
+// at compilation time instead of at runtime.
+#define MIGRAPHX_TEST "test"
+#define MIGRAPHX_TXT ".txt"
+#define MIGRAPHX_BZ2 ".bz2"
+
+#ifdef _WIN32
+#define MIGRAPHX_EXE_POSTFIX ".exe"
+#define MIGRAPHX_LIB_PREFIX
+#define MIGRAPHX_DYNAMIC_POSTFIX ".dll"
+#define MIGRAPHX_STATIC_POSTFIX ".lib"
+#define MIGRAPHX_OBJECT_POSTFIX ".obj"
+#else
+#define MIGRAPHX_EXE_POSTFIX
+#define MIGRAPHX_LIB_PREFIX "lib"
+#define MIGRAPHX_DYNAMIC_POSTFIX ".so"
+#define MIGRAPHX_STATIC_POSTFIX ".a"
+#define MIGRAPHX_OBJECT_POSTFIX ".o"
```

Comment:
These can be define as constants instead of macros.

Reply:
The goal is to concatenate at the compilation time, not the runtime.
For example, the shared object file name on Linux is `libsomething.so`, and the same dynamic library on Windows is named `something.dll`. By defining those macros, I can do `MIGRAPHX_LIB_PREFIX "something" MIGRAPHX_DYNAMIC_POSTFIX` which will result in the respective library names.
When I define them as constants, e.g., as `const char *`, I need to concatenate the string at runtime. There is a way of concatenating `std::string_view` at compile time, but it requires variable templated structures. I don't see any benefit here.

Reply:
This is for tests, I dont see any benefit in concating at compile time.